### PR TITLE
Building Extensions Guides

### PR DESCRIPTION
**Pull request by phazonoverload**
This PR introduces:

1. A load of new extension building guides.
2. A guides index page.
3. Guides sub-index pages for each category (also in sidebar).
4. The removal of the current Flows guides from the navigation (but they still exist).

2023-08-08T2200 33
2023-08-08T2200 47

[Deploy preview](https://deploy-preview-18952--directus-docs.netlify.app/guides)

### CODE REVIEW

**Comment by phazonoverload** [comment]
Waiting on just a couple more guides, but this should be it.

File: docs/guides/extensions/interfaces-relational-dropdown.md [hunk 1]
```diff
@@ -0,0 +1,651 @@
...
+description: Learn how to create a custom searchable dropdown with items from another collection.
+contributors: Tim Butterfield, Kevin Lewis
+---
+
+# Create A Searchable Dropdown With Items From Another Collection
+
+Interfaces provide a meaningful way for users to provide data. This guide will show you how to create a dropdown menu
+that is searchable for data from another collection.
+
+![A dropdown showing a number of values](https://marketing.directus.app/assets/46b4070d-fd77-4ee3-b08b-654519dda031)
```
Comment (en-jschuetze) [inline docs/guides/extensions/interfaces-relational-dropdown.md:11]: @phazonoverload the marketing.directus.app urls are note reachable (at least for me)

Comment (phazonoverload) [reply docs/guides/extensions/interfaces-relational-dropdown.md:11]: Fixed :)

File: docs/guides/extensions/operations-npm-package.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+icon: 'electric_bolt',
+description: 'Use Lodash Camel Case Function.',
+```
+
+Make sure the `id` is unique between all extensions including ones created by 3rd parties - a good practice is to
+include a professional prefix. You can choose an icon from the library [here](https://fonts.google.com/icons).
+
+With the information above, the operation will appear in the list like this:
+
+<img src="https://marketing.directus.app/assets/8e5245cb-2454-4c46-8a59-04a325385d61.png" alt="Add comment - add a comment to a record. A chat icon is displayed in the box." style="padding:3px 2px 2px;"/>
```
Comment (DanielBiegler) [inline docs/guides/extensions/operations-npm-package.md:52]:
```suggestion
<img src="https://marketing.directus.app/assets/8e5245cb-2454-4c46-8a59-04a325385d61.png" alt="Custom operation featuring the previously set icon, name and description" style="padding:3px 2px 2px;"/>
```

File: docs/guides/extensions/operations-npm-package.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+```
+
+## Add Operation to Directus
+
+In order to use this operation in Directus, you must copy the compiled index file into the project's extension folder.
+
+1. In the Directus extensions directory, open the operations directory and make a new directory called
+   `directus-operation-lodash`.
+2. From the operation's directory, open the **dist** folder and copy the `app.js` and `api.js` files into the directory.
+3. Install `lodash` helper library into your Directus root directory with `npm install lodash`.
```
Comment (DanielBiegler) [inline docs/guides/extensions/operations-npm-package.md:108]: The dependencies should have gotten bundled with the build-step, making this unnecessary.

**Review by DanielBiegler — CHANGES_REQUESTED** [review]

**Review by Nitwel — CHANGES_REQUESTED** [review]
Will continue reviewing tomorrow.

File: docs/guides/extensions/interfaces-radio-selector-icons.md [hunk 4]
```diff
@@ -0,0 +1,775 @@
...
+]
+```
+
+Add a field or fields for an icon. For the most flexibility, include SVG, Image and an Icon Picker.
+
+```js
+fields: [
+	{},
+	{},
+	{},
```
Comment (Nitwel) [inline docs/guides/extensions/interfaces-radio-selector-icons.md:124]:
```suggestion
	{...},
	{...},
	{...},
```
Could make it even clearer that these are not empty objects.

**Review by Nitwel — CHANGES_REQUESTED** [review]
Generally I'm not sure if we might want to use Vue script setup code style later on because it is the most up to date syntax. Same goes for using async await rather than `then` and `catch` in a few places. And a few images where a bit yellow, not sure if that was on purpose or not.

Other than these minor things, looking really good!!

File: docs/guides/extensions/layouts-getting-started.md [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+
+```
+npm i create-directus-extension
+npm init directus-extension
+```
+
+A list of options will appear (choose layout), and type a name for your extension (for example,
+`directus-layout-example`). For this guide, select JavaScript.
+
+Now the layout has been created, go into the current directory and build the extension. This must be performed whenever
```
[the remark this answers is not in the document]
Comment (phazonoverload) [reply docs/guides/extensions/layouts-getting-started.md:25]: `Now that the layout`, without the comma ;)

**Comment by phazonoverload** [comment]
Yeah the screenshots were provided a bit yellow. I think I can bundle it into the other items you've mentioned as a post-release update.

**Pull request merged**